docs/build-windows.md

```diff
@@ -5,8 +5,10 @@ Windows builds can be created using:
 - Visual Studio 2022 IDE suite with Clang/LLVM compiler, and vcpkg to provide
   dependencies. This is the fully-supported toolchain used to create release
   builds.
-- The Clang or GCC compilers using the Meson buildsystem running within the
-  MSYS2 environment to provide dependencies.
+
+- The Clang or GCC compilers using the Meson build system running within the
+  MSYS2 environment to provide dependencies (deprecated, support will be probably
+  removed in the future).
```

Comment:
One interpretation of **_'support being removed'_** is that "the supported thing" will no longer be possible from the user's perspective. 

On the dev side, that means removing the tangible stuff that was supporting the thing: be it workflows, build system code, or code-base features.

For example, if Microsoft says support for WIN32 binaries is being removed from Visual Studio 2026, the implication is it won't be able to compile WIN32 binaries.

But when I check our repo:
 - There are no MSYS2-specific workflows in CI
 - There is no MSYS2-specific code in Meson
 - There is no MSYS2-specific code in the source files

So I don't think we'll be actively deleting stuff to eliminate the ability to compile under MSYS2.

Instead, we probably mean that **_MSYS2 is not actively supported_**, the key being that we're not "removing" stuff. 

Well - I suppose the build notes themselves might be deleted, however the underlying ability is still possible.  So worst case, we could move the MSYS notes into a `build-msys2.md` file and mark it "not actively supported/tested; maintainers welcome!".

(The key here is that there's no MSYS2 stuff to maintain, the only fixes would be to the notes themselve or generic code-base fixes, because the code base has zero MSYS2 branches and would instead follow the same path as Visual Studio, which also uses Clang).

Reply:
>  The key here is that there's no MSYS2 stuff to maintain

Yes and no.  There's no specific MSYS2 stuff because it uses the same build system as Linux but with Windows headers.  In practice, it requires maintenance or stuff breaks.  Off the top of my head:

- It's rolling release and more bleeding edge than Arch Linux even.  Dependency and toolchain updates have broken the build multiple times with no changes on our side.
- The Windows header file definitions are slightly different than MSVC.  I've seen this cause compilation failure and the need to change for example a constant that was missing a `#define` in MSYS2.
- There was a weird bug where GCC had a different sleep implementation than MSVC that led to severe performance degradation.  Only affected GCC on MSYS2.  This was fixed but still these oddball things are possible.

>  Instead, we probably mean that MSYS2 is not actively supported, the key being that we're not "removing" stuff.

All that said, yeah maybe that wording makes more sense.  I don't think there's anything we can remove.  The whole point of MSYS2 is the ability to compile arbitrary POSIX programs.

For what it's worth, I do sometimes compile on MSYS2 whenever I'm on Windows but that's not very often so any bugs or build breaks are likely to linger due to lack of testing.

Reply:
Yeah, as @weirddan455 said, probably it will be difficult to compile anything but main due to the rolling nature of MSYS2. Well, Linux has the same issues, sure, but we're solving that with vcpkg that pins dependecies for anything but truly system level libs.

Plus yeah, MSYS2 had other non-trivial problems too as @weirddan455 mentioned. All in all, we should not provide build instructions untested and unmaintained by the team because they will be most likely be broken, and that's wasting people's time. We also don't want take up the burden maintaining multiple build systems on Windows, so the logical conclusion is to remove the MSYS2 stuff altogether.

I can adjust this comment as per your below suggestion, the when we get a confirmation from @shermp whether he wants to singlehandedly maintain MSYS2 going forward, we'll decide its fate. But my current understanding is no one wants to keep MSYS2 around.

Reply:
Also, personally, I'd remove all the other probably stale build instructions too if nobody tested they work in the last 12 months (all the weird Linux variants).

Reply:
> * There are no MSYS2-specific workflows in CI

But we do:

https://github.com/dosbox-staging/dosbox-staging/blob/main/.github/workflows/windows-msys2.yml

Historical context: The MSYS2 workflow was probably the no.1 catalyst for revamping our whole build system and migrate to vcpkg because it caused so much pain to prop it up with weird hacks over time as it just kept breaking... 😅

Reply:
Good to know; thanks for this background info @weirddan455 and @johnnovak !

Reply:
Perfect use of the wiki; maybe users can share how to build under Steam or WSL.

Reply:
Yeah, I appreciate without the historical context this removal might look like an arbitrary decision @interloper98, but we have our reasons 😄 The future end goals is vcpkg on all platforms + CMake instead of Meson.

I also get where you're coming from regarding "negative affirmative" statements such as "support for X is removed"—it looks weird out of context, no doubt. The high-level end-goal here is just provide "positive affirmative" instructions: this is how to build the software on all three platforms, this is what the CI workflows and the core team does, so this is 100% supported "official" build method. And then stop there 😄

Reply:
We'll regarding only going with positive affirmations 👍

*"you didn't specifically say that HPUX 4.3 wasn't supported, therefore I demand support! :neckbeard:"* 😂

Definitely could be numberous if we had to list them all!

![Linux_Distribution_Timeline](https://github.com/user-attachments/assets/9b384eaf-5741-462c-8c83-a5b1b3ce7c3a)

Reply:
Wow, what a chart from hell @interloper98 !  😅 😈  Also, today I learned about the neckbeard emoji 🤣